onadata/libs/serializers/monthly_submissions_serializer.py

```diff
@@ -0,0 +1,35 @@
+"""
+Monthly submissions serializer
+"""
+from rest_framework import serializers
+
+
+class MonthlySubmissionsListSerializer(serializers.ListSerializer):
+
+    def to_representation(self, data):
+        result = super(MonthlySubmissionsListSerializer,
+                       self).to_representation(data)
+        states = [True, False]
+        result_dictionary = {}
+
+        def get_result():
+            for state in states:
+                for i in result:
+                    if i['xform__shared'] == state:
+                        label = 'public' if state else 'private'
+                        result_dictionary[label] = i['num_instances']
+
+        get_result()
```

Comment:
I see no reason for the `get_result()` function, the loop block could easily be not within a function and still work.